src/components/AllSpecies/AllSpecies.tsx

```diff
@@ -8,16 +8,64 @@ type AllSpeciesProps = {
   characters: ModifiedCharacter[]
 }
 
+type CharacterState = {
+  originalCharacters: ModifiedCharacter[]
+  modifiedCharacters: ModifiedCharacter[]
+  selectedValue: string
+}
+
+type Action = { type: 'filter'; event: CharacterState['selectedValue'] }
+
+type Key = 'mass' | 'height'
+
+const sort = (characters: ModifiedCharacter[], key: Key) =>
+  characters.sort((a, b) => (b[key] || 0) - (a[key] || 0))
+
+const characterReducer = (state: CharacterState, action: Action): CharacterState => {
```

Comment:
> I'd love to receive a feedback in AllSpecies.tsx. Especially I want to know where is the good place to put reducer functions.

I wouldn't mind having this reducer logic in this file as it's small enough and encapsulated in this domain. Though it's ok to leave them here, if you don't feel comfortable leaving them in this file then I'd advise you to make a custom hook by extracting the logic so that you can separate the logic part of this file and render part of it. It could be called something like; `useSortCharacters`